lib/validators/serialtiles.js

```diff
@@ -21,20 +22,73 @@ module.exports = function validateSerialtiles(opts, callback) {
     .on('tile', function(t) {
       if (errored) return;
       if (!t.buffer) return;
-      if (t.buffer.length <= limits.max_tilesize) return;
-      callback(invalid('Tile exceeds maximum size of ' + Math.round(limits.max_tilesize/1024) + 'k at z' + t.z + '. Reduce the detail of data at this zoom level or omit it by adjusting your minzoom.'));
-      errored = true;
-      read.close();
+
+      var gzipped = (tiletype.headers(t.buffer))['Content-Encoding'] === 'gzip';
```

Comment:
@rclark I added this to differentiate between pbfs and image types to determine what should be Gunzipped. Also, related to comment above.